### ISSUE TITLE
py_binary zipapp error on Windows under cmd

### ISSUE THREAD

**Issue by benvanik** [bazelbuild/bazel#2768]
I'm trying to run a py_binary (via bazel run as well as genrules) on Windows and getting a path error in the zipapp __main__.py that bazel is embedding. It works correctly under msys/cygwin.

The exact error as returned on bazel 0.4.5:
```
λ bazel run tools/actions:generate_compile_command
INFO: Found 1 target...
Target //tools/actions:generate_compile_command up-to-date:
  C:/users/ben/appdata/local/temp/_bazel_ben/4qe-y0h3/execroot/xrtl/bazel-out/local-fastbuild/bin/tools/actions/generate_compile_command
INFO: Elapsed time: 0.328s, Critical Path: 0.01s

INFO: Running command line: C:/users/ben/appdata/local/temp/_bazel_ben/4qe-y0h3/execroot/xrtl/bazel-out/local-fastbuild/bin/tools/actions/generate_compile_command
Traceback (most recent call last):
  File "C:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\users\ben\appdata\local\temp\_bazel_ben\4qe-y0h3\execroot\xrtl\bazel-out\local-fastbuild\bin\tools\actions\generate_compile_command\__main__.py", line 168, in <module>
  File "C:\users\ben\appdata\local\temp\_bazel_ben\4qe-y0h3\execroot\xrtl\bazel-out\local-fastbuild\bin\tools\actions\generate_compile_command\__main__.py", line 121, in Main
  File "C:\users\ben\appdata\local\temp\_bazel_ben\4qe-y0h3\execroot\xrtl\bazel-out\local-fastbuild\bin\tools\actions\generate_compile_command\__main__.py", line 105, in GetRepositoriesImports
WindowsError: [Error 123] The filename, directory name, or volume label syntax is incorrect: '\\\\?\\c:\\users\\ben\\appdata\\local\\temp\\Bazel.runfiles_qzxzfz\\runfiles/*.*'
ERROR: Non-zero return code '1' from command: Process exited with status 1.
```

The issue is the `os.listdir` in [GetRepositoriesImports](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/stub_template.txt#L105); a listdir of '....\\runfiles' fails with the error, while a listdir of '...\\runfiles\\' will succeed. I confirmed this in python:
```
>>> import os
>>> os.listdir("\\\\?\\c:\\users\\ben\\appdata\\local\\temp\\Bazel.runfiles_qzxzfz\\runfiles")
>>> os.listdir("\\\\?\\c:\\users\\ben\\appdata\\local\\temp\\Bazel.runfiles_qzxzfz\\runfiles\\")
```
On cygwin both return a valid listing, while under cmd with the same exact python binary the first throws error 123 and the second succeeds.

I don't know why this difference exists, however the fix seems like adding a `\\` suffix in GetRepositoriesImports is a sufficient workaround.

**Comment by meteorcloudy** [comment]
That's for reporting, I will look into this.

**Comment by meteorcloudy** [comment]
@benvanik This issue should already be fixed by dce45d3151742338057f88f0a459ff442e05726f from HEAD.

It was caused by not normalizing a path start with `\\?\`
```
WindowsError: [Error 123] The filename, directory name, or volume label syntax is incorrect: '\\\\?\\c:\\users\\ben\\appdata\\local\\temp\\Bazel.runfiles_qzxzfz\\runfiles/*.*'
```
